src/Http/RouteConfig.php

```diff
@@ -8,18 +8,55 @@
 
 final class RouteConfig
 {
+    /** @var string The mark to give the next route in the matching Regex */
+    private string $regexMark = 'a';
+
     public function __construct(
         /** @var array<string, array<string, \Tempest\Http\Route>> */
-        public array $routes = [],
+        public array  $staticRoutes = [],
```

Comment:
I wonder if we could rewrite it like so?

```php
private array $staticRoutes = [];
private array $dynamicRoutes = [];

__construct(array $routes = [])
{
    foreach ($routes as $route) {
        $this->addRoute($route);
    }
}
```

So that from the outside, it's all still one array?

Reply:
How would you imagine accessing the properties in the GenericRouter would look like? I am currently accessing these directly there.

Reply:
You could wrap them in a generator:

```php
function all(): Generator
{
    foreach ($this->staticRoutes as $route) {
        yield $route;
    }
    
    foreach ($this->dynamicRoutes as $route) {
        yield $route;
    }
}
```

Reply:
I had also thought about that. This would mean that the "all" method does not return them grouped by http-method though.